File: src/main/java/com/stcode/bootstrap/model/Grkszc.java
package com.stcode.bootstrap.model;

import lombok.Data;

import java.util.Date;
import java.util.List;

/**
 * 监督检查-日常检查-跨省转出人员信息查询
 * [grkszc -> 个人跨省转出]
 */
@Data
public class Grkszc {

    //个人id
    private String grid;
    //代办机构
    private String dbjg;
    //多选
    private String[] dbjgs;
    //统一社会信用代码 [组织机构代码]
    private String dwdm;
    //单位ID -> GR_YLJF_ZYD
    private String dwid;
    //单位名称 -> DW_XX
    private String dwmc;

    //公民身份证号码 -> 社会保障号码 ->GR_XX
    private String bzhm;
    //日期起
    private String datefrom;
    //日期止
    private String dateto;
    //转入地行政区划代码 -> 转入地行政区划 -> GR_JFPZ_LXH
    private String zrxzqh;
    //转出地行政区划代码 -> 转出地行政区划 -> GR_JFPZ_LXH
    private String zcxzqh;
    //姓名
    private String grname;
    //检查类型 ->
    private String shlx;


    /**
     * 返回
     */
    //性别 -> 性别  (dmm=0201) -> GR_XX
    private String xb;
    //账户类型 ->帐户类别  dmm=0313 -> GR_YLJF_ZYD
    private String zhlb;
    //转出地(原参保地址) -> 转出社保名称 -> GR_YLJF_ZYD
    private String zcsbmc;
    //转入地(新参保地址) -> 转入社保名称 -> GR_YLJF_ZYD
    private String zrsbmc;
    // 缴费凭证 -> 打印日期 -> GR_JFPZ
    private String dyrq;
    // 缴费凭证 -> 经办人 -> GR_JFPZ
    private String jbr;
    // 联系函 -> 办理日期 -> GR_JFPZ_LXH
    private String blrq;
    //联系函 -> 经办人 -> GR_JFPZ_LXH
    private String lxhjbr;
    //信息表 -> 是否上传网络 -> 部网上传状态 0 已上传  1和空 未上传 -> GR_YLJF_ZYD
    private String zt;
    //信息表 -> 生成日期 -> 信息表上传日期 -> GR_YLJF_ZYD
    private String xxbscrq;
    //信息表 -> 经办人 -> GR_YLJF_ZYD
    private String xxbjbr;

    //养老资金财务支付情况 -> 转移支出总额 没找到
    private String zyyyyy;

    //养老资金财务支付情况 -> 统筹基金转移额 -> 统筹基金转移总额 ->GR_YLJF_ZYD
    private String tcjjzyze;
    //本地参保期间个人账户存储额(元) - >养老资金财务支付情况 -> 个人账户转移额 ->个人账户转移总额 ->GR_YLJF_ZYD
    private String grzhzyze;

    //养老资金财务支付情况 -> 财务付款日期 -> 没找到
    private String cwfkrq;

    //养老资金财务支付情况 -> 财务付款经办人 -> 没找到
    private String cwfkjbr;


    /**
     * 分页
     */
    private String offset;
    private String limit;

    /**
     * 检查结果jcjg
     */
    //检查结果ID
    private String[] formIds;
    //个人ID
    private String[] grIds;
    //公司id
    private String[] dwids;
    //备注
    private String memo;
    //接收个人ID和单位id
    List<JcjgVo> jcjgs;

    //检查人
    private String jcr;
    //检查日期
    private String jcrq;
    //检查
    private String jc;
    //重新检查
    private String cxjc;
    //检查结果
    private String jcjg;
    //检查 id
    private String jcid;
    //检查审核人
    private String jcshr;
    //检查审核日期
    private String jcshrq;
    //检查审核
    private String jcsh;
    //检查审核结果
    private String jcshjg;
    //检查审核备注
    private String jcshbz;

    // 细单 基本养老参保缴费凭证 后续有时间抽取公共类，改为继承方式
    //姓名
    private String xm;
    // todo 个人编号 BXH原个人编号 ->Gr_jfpz_lxh
    private String bxh;
    //编号 Gr_jfpz_lxh
    private String gdbh;
    //户籍 户籍属地 GR_XX
    private String hjsd;
    //在本地参保起止时间 - 本地缴费起日期 GR_JFPZ
    private String qrq;
    //本地实际缴费月数 在本地实际缴费月数 GR_YLJF_ZYD
    private String bdjfys;

    //todo 行政区划代码 ->参保地行政区划代码 GR_YLJF_ZYD 字段无效
    private String xzqh;

    //电话 联系电话 ->GR_YLJF_ZYD
    private String lxdh;
    //地址 邮寄地址 ->GR_YLJF_ZYD
    private String yjdz;
    //邮编 居住地邮政编码 -> GR_XX
    private String yzbm;

    /**
     * 细单 基本养老保险缴费账户转移联系函
     */
    //原参保地社保机构名称 参保地社保机构名称 ->GR_JFPZ_LXH
    private String jfqxmc;

    //原在你处的参保人员 -> 参保人员姓名 GR_JFPZ_LXH
    private String cbryxm;
    //新就业地社保机构开户全称 GR_JFPZ_LXH
    private String khqc;
    //新就业地社保机构开户银行 GR_JFPZ_LXH
    private String khyh;
    //新就业地社保机构银行账号 GR_JFPZ_LXH
    private String yhzh;
    //新就业地社保机构地址 GR_JFPZ_LXH
    private String sbjgdz;
    //新就业地社保机构邮政编码 GR_JFPZ_LXH
    private String sbjgdzyzbm;
    //新就业地社保机构	新就业的社保机构章  汉字 GR_JFPZ_LXH
    private String newjfqxname;

    /**
     * 细单 参保人基本信息
     */
    //出生日期
    private String csrq;
    //参加工作时间 - >GR_YLJF_ZYD
    private String gzrq;
    //首次参保地实行个人缴费时间 ->  GR_YLJF_ZYD
    private String cbdgrjfrq;
    //本人首次缴费时间 ->GR_YLJF_ZYD
    private String scjfrq;
    //本人建立个人账户时间 GR_YLJF_ZYD
    private String grzhrq;
    //在本地缴费起始时间  GR_YLJF_ZYD
    private String bdjfqrq;
    //在本地缴费终止时间 GR_YLJF_ZYD
    private String bdjfzrq;
    //在本地实际缴费时间 - >在本地实际缴费月数 GR_YLJF_ZYD

    //转移日期 GR_YLJF_ZYD
    private String zyrq;
    //1998年1月1日前账户中个人缴费累计存储额1 -> 1998年1月1日前个人缴费累计存储额 GR_YLJF_ZYD
    private String grjf_98q;
    //1998年1月1日至调转上年末个人账户累计存储存额2 -> 1998年1月1日后个人账户累计存储额 GR_YLJF_ZYD
    private String grzh_98h;
    //调转当年计入个人账户本金金额3
    private String dngrzhbj;
    //个人账户基金转移额4 - > 个人账户转移总额 GR_YLJF_ZYD
    // private String grzhzyze;
    //统筹基金转移额5 ->统筹基金转移总额 GR_YLJF_ZYD
//    private String tcjjzyze;
//    转移基金总额6 ->转移基金总额  GR_YLJF_ZYD
    private String zyjjze;
    //行政区划名称
    private String xzmc;
    //年份 GR_YLJF_ZYD_JXMX
    private String nian;
    //缴费起止时间
    //月缴费基数 GR_YLJF_ZYD_JXMX
    private String yjfjs;
    //缴费月数 GR_YLJF_ZYD_JXMX
    private String ys;
    //缴费比例 单位 GR_YLJF_ZYD_JXMX
    private String dwjfbl;
    //缴费比例 划入个人账户比例 GR_YLJF_ZYD_JXMX
    private String dwhzbl;
    //缴费比例 个人 GR_YLJF_ZYD_JXMX
    private String grjfbl;
    //当年记账金额 小计
    private String dnjzje;
    //当年记账金额 个人缴费
    private String dngrjf;
    //当年记账利息 小计
    private String dnjzlx;
    //当年记账利息 个人缴费
    private String dngrlx;
    //至本年末账户累计存储额 小计
    private String zbnmljzhje;
    //至本年末账户累计存储额 个人缴费
    private String zbnmljgrjf;


    //是否全部检查，用于判断是本页检查还是全部检查
    private String checkFlag;

}

File: src/main/resources/static/js/grkszc/grkszc.js
$(function(){
    _grkszc.init();

    //主页面查询表格
    $("#query_btn").click(function(){
        $("#mainDataGrid").bootstrapTable("destroy");
        _grkszc.mainDataGrid();
    });


    //点击检查
    $("#check-btn").click(function () {
        //判断是否有值
        var isRows = $('#mainDataGrid').bootstrapTable('getData',false);
        if(isRows.length <= 0){
            bootbox.alert("没有检查记录!");
            return;
        }

        var rows =  $("#mainDataGrid").bootstrapTable('getSelections');
        if(rows.length <= 0){
            bootbox.confirm("是否对本次查询结果中【未检查】记录全部检查?",
                function(result) {
                    if (result) {
                        // 查询检查项 modalID:模态框ID,formID表单id,isCheck:检查和从新检查标记
                        getCheckResultTable("grkszcDetailCheck","grkszcDetailForm","/grkszc/checkResultTable",1);
                    }
                });
        }else {
            if(alreadyCheck(rows)){
                getCheckResultTable("grkszcDetailCheck","grkszcDetailForm","/grkszc/checkResultTable",2);
            }
        }
    });

    //点击重新检查
    $("#restart-check-btn").click(function () {
        if(cxcheck()){
            getCheckResultTable("grkszcDetailCheck","grkszcDetailForm","/grkszc/checkResultTable",3);
        }
    });


    //检查确定
    $("#check-enter-btn").click(function () {
        //获取选项检查结果
        var temp = getOptionCheckResult("grkszcDetailForm");

        if(checkFlag == 1 || checkFlag == 2){ //检查

            var rows =  $("#mainDataGrid").bootstrapTable('getSelections');
            if(rows.length <= 0){
                //获取当前全部
                rows = $('#mainDataGrid').bootstrapTable('getData',false);
            }
            if(rows.length <=0){
                bootbox.alert("没有数据，不能检查！");
                return false;
            }
            var jcjgtemp = getJCJGS(rows);
            var params = JSON.stringify({formIds:temp.formIds,memo:temp.memo,jcjgs:jcjgtemp.jcjgs,checkFlag:checkFlag});
            params =(JSON.stringify($("#mainform").serializeJSON()) + params).replace(/}{/, ',');

            $.ajax({
                type: "post",
                contentType:"application/json;charset=utf-8",
                dataType: "json",
                url: "/grkszc/insertJcjg",
                data: params,
                success: function(msg){
                    if (msg.code == 0) {
                        $("#grkszcDetailCheck").modal('hide');
                        bootbox.alert({
                            message: "检查完成,成功检查【"+msg.num+"】条。",
                            size: 'small'
                        });
                        $("#mainDataGrid").bootstrapTable('refresh');
                    }else {
                        bootbox.alert(msg.error);
                    }
                }
            });


        }else {//重新检查

            var rows =  $("#mainDataGrid").bootstrapTable('getSelections');
            var jcjgtemp = getJCJGS(rows);
            var params = JSON.stringify({formIds:temp.formIds,memo:temp.memo,jcjgs:jcjgtemp.jcjgs,checkFlag:checkFlag});

            $.ajax({
                type: "post",
                contentType:"application/json;charset=utf-8",
                dataType: "json",
                url: "/grkszc/updateJcjg",
                data: params,
                success: function(msg){
                    if (msg.code == 0) {
                        bootbox.alert({
                            message: "重新检查完成",
                            size: 'small'
                        });
                        $("#grkszcDetailCheck").modal('hide');
                        $("#mainDataGrid").bootstrapTable('refresh');
                    }else {
                        bootbox.alert(msg.error);
                    }
                }
            });
        }



    });


    //检查审核
    $("#examine-btn").click(function(){
        if(cxcheck()){
            getCheckResultTable("examine-modal","examineModalForm","/grkszc/getjcshTable",4);
        }

    });

    //审核检查确定
    $("#examine-enter-btn").click(function(){
        //获取选项检查结果
        var temp = getOptionCheckResult("examineModalForm");

        var rows =  $("#mainDataGrid").bootstrapTable('getSelections');
        var jcjgtemp = getJCJGS(rows);
        var params = JSON.stringify({formIds:temp.formIds,memo:temp.memo,jcjgs:jcjgtemp.jcjgs,checkFlag:checkFlag});

        $.ajax({
            type: "post",
            contentType:"application/json;charset=utf-8",
            dataType: "json",
            url: "/grkszc/examineCheckJcjg",
            data: params,
            success: function(msg){
                if (msg.code == 0) {
                    bootbox.alert({
                        message: "检查审核完成",
                        size: 'small'
                    });
                    $("#examine-modal").modal('hide');
                    $("#mainDataGrid").bootstrapTable('refresh');
                }else {
                    bootbox.alert(msg.error);
                }
            }
        });
    });

    //设置下拉框宽度
    $("#dbjg").selectpicker({
        "width":250,
        noneSelectedText:'请选择'
    });

});

//全局变量 1:检查,全部检查，2:检查，本页检查，3:重新检查，
var checkFlag     = 0;

//参保人员基本信息
function grPayDetail(value , record , index) {
    return '<span id="hoverspan" onclick="_grkszc.grDetailInfo('+index+')">' + value + '</span>';
}
//基本养老保险缴费账户转移联系函 详情
function grlxhDetail(value , record , index) {
    return '<span id="hoverspan" onclick="_grkszc.grlxhInfo('+index+')">' + value + '</span>';
}
//信息表
function grxxbDetail(value , record , index) {
    return '<span id="hoverspan" onclick="_grkszc.grxxbInfo('+index+')">' + value + '</span>';
}

//转换检查
function formatIsJC(value, row, index) {
    return value == 'true' ? "是" : "否";
}

//是否重新检查
function formatIsCXJC(value, row, index) {
    return value == 'true' ? "是" : "否";
}


//设置表单值
$.fn.setForm = function(jsonValue){
    var obj = this;
    $.each(jsonValue,function(name,ival){
        obj.find("input[name="+name+"]").val(ival)
        obj.find("[name="+name+"]").text(ival);
    })
}



// 查询检查项 modalID:模态框ID,formID表单id,isCheck:检查和从新检查标记
function getCheckResultTable(modalID,formID,url,isCheck){
    checkFlag = isCheck;
    $.ajax({
        type: "post",
        url: url,
        data: "",
        success: function(msg){
            if (msg.code == 0) {
                $("#"+formID).find("tbody").empty();
                var datalist = msg.data;
                $.each(datalist,function(index,item){
                    $("#"+formID).find("tbody").append("<tr><td><div class=\"checkbox\"  style=\"text-align: center;margin:auto;\"><input type=\"checkbox\" name='"+item.mxdm+"' value='"+item.dmmxid+"'/></div></td><td>"+item.mxmc+"</td></tr>");
                });
                $("#"+formID).find("tbody").append("<tr><td><div style=\"text-align: center;margin:auto;\">备注</div></td><td><textarea style='resize: none' class=\"form-control\" type=\"text\" name=\"memo\"></textarea></td></tr>");
            } else {
                bootbox.alert(msg.error);
            }
            $("#"+modalID).modal('show');
        }
    });

}

//检查前判断
function alreadyCheck() {
    var rows =  $("#mainDataGrid").bootstrapTable('getSelections');
    if(rows.length <= 0){
        bootbox.alert("请选择要检查的记录!");
        return false;
    }

    var flag = false;
    //1、看是否已经检查了
    $.each(rows,function(index,item){
        if(item.jc == "true"){
            flag =true;
            return false;
        }
    });
    if(flag){
        flag = false;
        bootbox.alert("您选择的记录是已经检查的!");
        return false;
    }
    return true;
}

//获取选择检查结果
function getOptionCheckResult(id){
    var formIds         = new Array();
    var memo;
    //取提交的
    var formdata    =  $("#"+id).serializeArray();
    $.each(formdata,function(index,item){
        if(item.name == "memo"){
            memo = item.value;
        } else {
            formIds.push(item.value)
        }
    });

    var result = {
        "formIds":formIds,
        "memo":memo
    }
    return result;
}

function getJCJGS(rows){
    //存放jcjg集合
    var jcjgs           = new Array();
    for(var i=0;i<rows.length;i++){
        var jcjg = {};
        jcjg.grid = rows[i].grid;
        jcjg.dwid = rows[i].dwid;
        jcjg.jcid = rows[i].jcid;
        jcjgs.push(jcjg);
    }

    var result = {
        "jcjgs":jcjgs
    }
    return result;
}

//重新检查验证
function cxcheck() {
    //1、选择含有未检查的给予提示
    var rows =  $("#mainDataGrid").bootstrapTable('getSelections');
    if(rows.length <= 0){
        bootbox.alert("请选择已检查的记录");
        return false;
    }
    var flag = false;
    //2、选有检查过的，按钮可用
    $.each(rows,function(index,item){
        if(item.jc == "false" || item.jc == null){
            flag =true;
            return false;
        }
    });
    if(flag){
        flag = false;
        bootbox.alert("您选择的记录包含未检查的记录！");
        return false;
    }
    return true;
}


_grkszc = {
    init: function () {
        this.mainDataGrid();
    },

    //缴费凭证 显示个人信息
    grDetailInfo:function (index) {
        //获取行号
        var record =  $('#mainDataGrid').bootstrapTable("getData")[index];
        $.ajax({
            type: "post",
            url: "/grkszc/grDetailInfo",
            data: {grid: record.grid,dwid:record.dwid},
            success: function(msg){
                $("#payDetailForm").setForm(msg.data);
                $("#payDetail").modal('show');
            }
        });

    },
    //点击姓名获取，联系函详情
    grlxhInfo:function (index) {
        //获取行号
        var record =  $('#mainDataGrid').bootstrapTable("getData")[index];
        $.ajax({
            type: "post",
            url: "/grkszc/grlxhInfo",
            data: {grid: record.grid,dwid:record.dwid},
            success: function(msg){
                $("#lxhForm").setForm(msg.data);
                $("#lxhModal").modal('show');
            }
        });

    },
    //点击信息表，姓名
    grxxbInfo:function (index) {
        //获取行号
        var record =  $('#mainDataGrid').bootstrapTable("getData")[index];
        $.ajax({
            type: "post",
            url: "/grkszc/grxxbInfo",
            data: {grid: record.grid,dwid:record.dwid},
            success: function(msg){
                $("#xxbForm").setForm(msg.data);
                $("#xxbgrcx").modal('show');
                $("#xxbDataGrid").bootstrapTable("destroy");
                _grkszc.xxbDataGrid(record);
            }
        });
    },

    mainDataGrid:function () {
        $('#mainDataGrid').bootstrapTable({
            method: 'post',
            url: "/grkszc/search",
            striped: true,
            pagination: true,
            sidePagination: 'server',
            pageSize: 5,
            contentType:"application/x-www-form-urlencoded",
            cache: false,
            pageList: [5, 25, 50, 100],
            queryParams: function(params) {
                var formdata    = $("#mainform").serialize();
                var paging      = formdata+"&offset="+params.offset+"&"+"limit="+params.limit;
                return paging;
            },

            columns: [

                [{"title": "基本养老保险转出情况", "halign": "center", "align": "center", "colspan": 33}],
                [{checkbox: true,colspan: 1, rowspan: 2},
                    {title: 'grid', field: 'grid', visible: false, colspan: 1, rowspan: 2},
                    {title: 'jcid', field: 'jcid', visible: false, colspan: 1, rowspan: 2},
                    {title: '序号', align: 'center', valign: 'middle', width: 100, colspan: 1, rowspan: 2,
                        formatter: function(value, row, index) {
                            var pageSize    = $('#mainDataGrid').bootstrapTable('getOptions').pageSize;
                            var pageNumber  = $('#mainDataGrid').bootstrapTable('getOptions').pageNumber;
                            return pageSize * (pageNumber - 1) + index + 1;
                        }
                    },
                    {title: '统一社区信用代码',    field: 'dwdm',         align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2, width: 100},
                    {title: '单位名称',            field: 'dwmc',         align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2, width: 100},
                    {title: '社会保障号码',        field: 'bzhm',         align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2, width: 100},
                    {title: '姓名',                field: 'grname',       align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2, width: 100},
                    {title: '性别',                field: 'xb',           align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2, width: 100},
                    {title: '账户类型',            field: 'zhlb',         align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2, width: 100},
                    {title: '转出地(原参保地址)',   field: 'zcsbmc',      align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2, width: 100},
                    {title: '转入地(新参保地)',     field: 'zrsbmc',      align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2, width: 100},
                    {title: '缴费凭证',             field: 'jfpz',        align: 'center',    valign: 'middle',  colspan: 2, rowspan: 1, width: 100},
                    {title: '联系函',               field: 'lxh',         align: 'center',    valign: 'middle',  colspan: 2, rowspan: 1, width: 100},
                    {title: '信息表',               field: 'xxb',         align: 'center',    valign: 'middle',  colspan: 3, rowspan: 1, width: 100},
                    {title: '养老资金财务支付情况',  field: 'ylzjzfqk',    align: 'center',    valign: 'middle',  colspan: 5, rowspan: 1, width: 100},
                    {title: '检查人',               field: 'jcr',          align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2, width: 100},
                    {title: '检查日期',             field: 'jcrq',        align: 'center',    valign: 'middle', sortable: true, colspan: 1, rowspan: 2, width: 100},
                    {title: '检查',                 field: 'jc',          align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2, width: 100,formatter: formatIsJC},
                    {title: '重新检查',             field: 'cxjc',        align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2, width: 100,formatter: formatIsCXJC},
                    {title: '备注',                 field: 'memo',        align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2, width: 100},
                    {title: '检查审核人',           field: 'jcshr',       align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2, width: 100},
                    {title: '检查审核日期',         field: 'jcshrq',      align: 'center',    valign: 'middle', sortable: true, colspan: 1, rowspan: 2, width: 100},
                    {title: '检查审核',             field: 'jcsh',      align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2, width: 100},
                    {title: '检查审核结果',         field: 'jcshjg',      align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2, width: 100},

                ],
                [
                    {field: 'dyrq', title: '打印日期',            align: 'center', valign: 'middle',width: 100, sortable: true,formatter: grPayDetail},
                    {field: 'jbr',   title: '经办人',             align: 'center', valign: 'middle',width: 100, sortable: true},
                    {field: 'lxhblrq',   title: '办理日期',       align: 'center', valign: 'middle',width: 100, sortable: true},
                    {field: 'lxhjbr', title: '经办人',            align: 'center', valign: 'middle',width: 100, sortable: true,formatter: grlxhDetail},
                    {field: 'xxbscrq', title: '生成日期',         align: 'center', valign: 'middle',width: 100, sortable: true},
                    {field: 'xxbjbr', title: '经办人',            align: 'center', valign: 'middle',width: 100, sortable: true,formatter:grxxbDetail},
                    {field: 'zt', title: '是否上传网络',          align: 'center', valign: 'middle',width: 100, sortable: true},
                    {field: 'zyyyyy', title: '转移支出总额',      align: 'center', valign: 'middle',width: 100, sortable: true},
                    {field: 'tcjjzyze', title: '统筹基金转移额',  align: 'center', valign: 'middle',width: 100, sortable: true},
                    {field: 'grzhzyze', title: '个人账户转移额',  align: 'center', valign: 'middle',width: 100, sortable: true},
                    {field: 'cwfkrq', title: '财务付款日期',      align: 'center', valign: 'middle',width: 100, sortable: true},
                    {field: 'cwfkjbr', title: '财务付款经办人',   align: 'center', valign: 'middle',width: 100, sortable: true}
                ]
            ]

        });
    },

    xxbDataGrid:function (data) {
        $('#xxbDataGrid').bootstrapTable({
            method: 'post',
            url: "/grkszc/searchXxb",
            striped: true,
            pagination: true,
            sidePagination: 'server',
            pageSize: 5,
            contentType:"application/x-www-form-urlencoded",
            cache: false,
            pageList: [5, 25, 50, 100],
            queryParams: function(params) {
                var paging = {
                    offset: params.offset,
                    limit: params.limit,
                    grid:data.grid,
                }
                return paging;
            },

            columns: [

                [
                    {title: 'grid', field: 'grid', visible: false, colspan: 1, rowspan: 2},
                    // {title: '序号', align: 'center', valign: 'middle', width: 100, sortable: true,colspan: 1, rowspan: 2,
                    //     formatter: function(value, row, index) {
                    //         var pageSize    = $('#xxbDataGrid').bootstrapTable('getOptions').pageSize;
                    //         var pageNumber  = $('#xxbDataGrid').bootstrapTable('getOptions').pageNumber;
                    //         return pageSize * (pageNumber - 1) + index + 1;
                    //     }
                    // },
                    {title: '参保地区',                field: 'cbdq',        align: 'center',    valign: 'middle',  colspan: 2, rowspan: 1,width:100},
                    {title: '年份',                    field: 'nian',        align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2,width:60},
                    {title: '缴费起止时间',            field: 'qrq',         align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2,width:100},
                    {title: '缴费月数',                field: 'ys',          align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2,width:50},
                    {title: '月缴费基数',              field: 'yjfjs',       align: 'center',    valign: 'middle',  colspan: 1, rowspan: 2,width:100},
                    {title: '缴费比例',                field: 'jfbl',        align: 'center',    valign: 'middle',  colspan: 3, rowspan: 1,width:100},
                    {title: '当年记账金额',            field: 'dnjzje',      align: 'center',    valign: 'middle',  colspan: 2, rowspan: 1,width:100},
                    {title: '当年记账利息',            field: 'dnjzlx',      align: 'center',    valign: 'middle',  colspan: 2, rowspan: 1,width:100},
                    {title: '至本年末账户累计存储额',   field: 'zbnmzh',      align: 'center',   valign: 'middle',  colspan: 2, rowspan: 1,width:100},
                    {title: '备注',                    field: 'memo',        align: 'center',   valign: 'middle',  colspan: 1, rowspan: 2,width:50},
                ],
                [
                    {field: 'xzqh',         title: '行政区划代码',        align: 'center', valign: 'middle', formatter: grPayDetail,width:100},
                    {field: 'xzmc',         title: '名称',                align: 'center', valign: 'middle',width:200},
                    {field: 'dwjfbl',       title: '单位',                align: 'center', valign: 'middle',width:100},
                    {field: 'dwhzbl',       title: '划入个人账户比例',     align: 'center', valign: 'middle', formatter: grlxhDetail,width:100},
                    {field: 'grjfbl',       title: '个人',                align: 'center', valign: 'middle',width:100},
                    {field: 'dnjzje',       title: '小计',                align: 'center', valign: 'middle', formatter:grxxbDetail,width:100},
                    {field: 'dngrjf',       title: '个人缴费',            align: 'center', valign: 'middle',width:100},
                    {field: 'dnjzlx',       title: '小计',                align: 'center', valign: 'middle',width:100},
                    {field: 'dngrlx',       title: '个人缴费',            align: 'center', valign: 'middle',width:100},
                    {field: 'zbnmljzhje',   title: '小计',                align: 'center', valign: 'middle',width:100},
                    {field: 'zbnmljgrjf',   title: '个人缴费',            align: 'center', valign: 'middle',width:100},
                ]
            ]

        });
    },

    isNull:function(a) {
        if (a == null || a == undefined || (typeof a == 'string' && a == '')) {
            return true;
        }
        return false;
    },

}

